### DISCUSSION TITLE
Using the PyTorch API, how to save and load models?

### DISCUSSION THREAD

**Discussion by zhuyuhang4** [AdaptiveMotorControlLab/CEBRA#130] [Q&A] [answered]
Using the PyTorch API, how to save and load models?

**Comment by stes** [comment, marked as answer]
Hi @zhuyuhang4 ,

multiple options.

If you want to implement saving/loading logic your self, [`torch.save`](https://pytorch.org/docs/stable/generated/torch.save.html) and [`torch.load`](https://pytorch.org/docs/stable/generated/torch.load.html) or pretty much any other serialization approach (e.g. pickle, joblib, ...) will do. You should save model, optimizer, relevant arguments, etc.

To make this a bit more convenient, we implemented [loading](https://cebra.ai/docs/api/pytorch/solvers.html#cebra.solver.base.Solver.load) and [saving](https://cebra.ai/docs/api/pytorch/solvers.html#cebra.solver.base.Solver.save) also directly in the `Solver` class.

The functions essentially take the [`state_dict`](https://cebra.ai/docs/api/pytorch/solvers.html#cebra.solver.base.Solver.state_dict) of the solver you are running and serialize these.

If this answers your Q, please mark as an answer :) Otherwise, happy to help further.
